### ISSUE TITLE
KotlinTest not harmonious with IntelliJ IDEA

### ISSUE THREAD

**Issue by russel** [kotlintest/kotlintest#39]
Writing KotlinTest tests in IntelliJ editor works fine. Executing individual KotlinTest files in IntelliJ runs the tests. However trying to run "All Tests" in the test module of an IntelliJ IDEA project with KotlinTest tests results in no tests being found.

**Comment by sksamuel** [comment]
It seemed to work for me and these are just standard jUnit tests under the hood. @helmbold does it work for you ?

**Comment by helmbold** [comment]
It works fine here. The only thing I'm missing is the possibility to run a single test method, but I guess that this would require a KotlinTest plug-in for IntelliJ.

**Comment by russel** [comment]
Here's an hypothesis: @sksamuel and @helmbold have their tests in the same directory as the source being tested and created their projects using IntelliJ IDEA whereas I have the standard Gradle project structure with source in src/main/kotlin and tests in src/test/kotlin using an IntelliJ IDEA project generated from the Gradle project.

**Comment by helmbold** [comment]
Let's use KotlinTest itself as an example: it has the code separated from tests (src/main/kotlin and src/test/kotlin). It [works on my machine](https://blog.codinghorror.com/the-works-on-my-machine-certification-program/).

**Comment by sksamuel** [comment]
Me too. In the left hand side "project" pane, I right click on src/test/kotlin (the green icon) and choose "run all tests". Works fine. Is that what you're doing Russel

**Comment by russel** [comment]
Part of the hypothesis failed then: file system and module structure not the issue. Run "gradle test" and all the tests run. Run the files individually in IntelliJ IDEA and they run. Right click on package or module and "No tests found". Recreating the project from scratch leads to the same position. Are you both creating projects with the built in Kotlin support rather than with Gradle, if so the the remaining hypothesis is that the Gradle plugin is doing something really weird.

**Comment by sksamuel** [comment]
I go to file -> new project from existing sources -> ... import from gradle

I'm running intellij 16 community.

**Comment by sksamuel** [comment]
on Linux.

**Comment by russel** [comment]
I am running 2016.2 EAP and use the import route it should do the same thing…

**Comment by helmbold** [comment]
@russel Maybe you should file an issue, since it looks like some IntelliJ bug. Could you try IntelliJ 2016.1?

**Comment by sksamuel** [comment]
Did you get this sorted  @russel

**Comment by mfriedenhagen** [comment]
I see the same behavior as @russel, so I opened https://youtrack.jetbrains.com/issue/IDEA-158136.

**Comment by russel** [comment]
@mfriedenhagen Thanks for that, I have voted. Sorry for not acting myself, I have been writing a document and avoiding as much as possible. Document submitted this morning, so no I can get back to doing stuff. Which will probably be kotlinx.maths code using KotlinTest tests.

**Comment by hoangtranwork** [comment]
I got the same issue on elementaryOS, IDEA 2016.2 EAP. Here a screenshot of runing all test in this repo's master: https://www.dropbox.com/s/k1c5bsd3e0yfcov/Screenshot_2016-07-11_16.50.10.png

**Comment by hoangtranwork** [comment]
ah, just found a workaround by learning from Spek code: adding `@RunWith()` to our test class:
```
@RunWith(KTestJUnitRunner::class)
class FlatSpecExample : FlatSpec() {
    init {
        "String.length" should "return the length of the string" {
            "sammy".length shouldBe 5
            "".length shouldBe 0
        }
    }
}
```
and then our test class will be picked up by IDEA's Junit Runner

**Comment by russel** [comment]
Aha. the `@RunWith` thing is working for me.

**Comment by mfriedenhagen** [comment]
Well, TestBase is annotated with the RunWith, so this seems not to propagate.

**Comment by helmbold** [comment]
I think I've fixed it (for 2.0 and 1.3). Could you verify it please, @russel , @mfriedenhagen , @hoangtranwork ?

**Comment by russel** [comment]
Is there a released version of KotlinTest available with the fix.

(This would be amazingly useful for session at the LJC Open Conference on Saturday 2016-11-26.)

**Comment by sksamuel** [comment]
I'll do a release tomorrow night. Remind me :)

**Comment by helmbold** [comment]
Since nobody reports any issues with my solution, I close it now.

**Comment by russel** [comment]
@helmbold As I failed to remind @sksamuel to do a release to test, I haven't actually tested. But the opportunity I had for last Saturday has gone, and went differently to expected. People did though get a blast of proposition based testing and because I have no idea if it is sensible using the various Java/JUnit ports of QuickCheck, I showed using Kotlin and KotlinTest. Speks got a mention from the audience. Hopefully people spotted that Kotlin is way nicer than Java.

**Comment by helmbold** [comment]
@russel It is great that you did something to improve our visibility!

**Comment by elect86** [comment]
I still need `@RunWith(KTestJUnitRunner::class)`

Idea 2016.3.1
Kotlin 1.1 m03
kotlintest 1.3.5

**Comment by helmbold** [comment]
Thanks for the feedback, @elect86 . This issue is fixed in 2.0 (dev) only at the moment. We should get this release finally out of the door!

**Comment by MrTact** [comment]
For anyone who arrives at this issue via searching for solutions the problem, as I did, a humble reminder: if you added the test/kotlin directory by hand, don't forget that you have to do "Mark directory as test root."
